compatible with Python 3.12

piranna opened the pull request:
Fix #49

piranna committed:
Remove useless file

piranna committed:
Remove code for project version to make compatible with Python 3.12

kra3 commented on macaddress/__init__.py:
```diff
@@ -44,15 +44,3 @@ def format_mac(eui_obj, dialect):
-    __version__ = 'Please install this project with setup.py'
-else:
-    __version__ = _dist.version
-VERSION = __version__   # synonym
```
VERSION was added as per django-macaddress/django-macaddress/issues/7 (to provide version visible in Django--debug-toolbar)

removing it would be a regression, rather fix should be replacing the pkg_resource call with it's replacement

kra3 replied:
From the pkg_resources docs:https://setuptools.pypa.io/en/latest/pkg_resources.html

>Use of pkg_resources is deprecated in favor of [importlib.resources](https://docs.python.org/3.11/library/importlib.resources.html#module-importlib.resources), [importlib.metadata](https://docs.python.org/3.11/library/importlib.metadata.html#module-importlib.metadata) and their backports ([importlib_resources](https://pypi.org/project/importlib_resources), [importlib_metadata](https://pypi.org/project/importlib_metadata)). Some useful APIs are also provided by [packaging](https://pypi.org/project/packaging) (e.g. requirements and version parsing). Users should refrain from new usage of pkg_resources and should work to port to importlib-based solutions.

Python 3.12 has removed pkg_resources from the standard library (moved to setuptools):
https://docs.python.org/3/whatsnew/3.12.html
> https://github.com/python/cpython/issues/95299: Do not pre-install setuptools in virtual environments created with [venv](https://docs.python.org/3/library/venv.html#module-venv). This means that distutils, setuptools, pkg_resources, and easy_install will no longer available by default; to access these run pip install setuptools in the [activated](https://docs.python.org/3/library/venv.html#venv-explanation) virtual environment.

kra3 replied:
here is one way of addressing this without additional libraries: https://github.com/googleapis/python-api-core/pull/80

piranna replied:
So, do you want i update the PR with this?

piranna committed:
Clean-up

piranna committed:
Replace invalid `assertEquals` for `assertEqual`

piranna committed:
Restored `version`, put on its own file

piranna commented:
Any update on this? Can we get it merged?

piranna committed:
Updated Python version to minimum required by `netaddr` dependency

piranna committed:
Add `.coverage` file to `.gitignore` and removed it

piranna committed:
Migrated to `pyproject.toml`

piranna committed:
Enable Poetry package mode & rename package

piranna committed:
Fix `license` and remove `long_description` fields

piranna committed:
Clean-up

piranna committed:
Get directly version string

kra3 commented:
thanks a lot for addressing the concerns and upgrading the build tooling to poetry. 

I have also realised travis integration is broken, so I've replaced travis with GitHub workflows (#45). 
Could you please kind enough to rebase the PR branch to main branch. Once the build pass, I will cut a release.

piranna commented:
> thanks a lot for addressing the concerns and upgrading the build tooling to poetry.

You are welcome.


> Could you please kind enough to rebase the PR branch to main branch.

Done, can you review it?

kra3 commented:
unfortunately, tests are failing here (needs to update workflows to use poetry). 
However  test are failing for me locally `poetry run pytest`. 
Ware you able to run tests successfully in your local machine ? how are you running tests ?

Djailla commented:
Duplicated from last release, can be closed
